### ISSUE TITLE
Edit settings without kernel detatch

### ISSUE THREAD

**Issue by Xeue** [vijay-prema/audient-evo-linux-tools#4]
Hi, great work with this, glad to see it working!
The detaching and re-attaching to the kernel causes me some issues.
I saw from the other issues you no longer have this device, but I wondered if you could point me in a direction as to how you would go about making this work without detaching and re-attaching.
I'm not massively experienced with python so not quite sure where I'd start....

**Comment by vijay-prema** [comment]
Yes I wont be able to help much sorry, due to no longer having the device.

My experience with pyusb is limited and the docs are not that good.  You can find some documentation within the code itself e.g. the code [comments in the code](https://github.com/pyusb/pyusb/blob/master/usb/core.py) for the kernel related functions.  There is also the pyusb [tutorial](https://github.com/pyusb/pyusb/blob/master/docs/tutorial.rst) but it doesn't talk about those functions.

My understanding was that I needed to use those detach/re-attach functions in order to send the control messages reliably from a user python script, in case the device was already in use by a kernel driver (e.g. maybe device is already capturing audio via kernel driver).  Of course that could interfere with the kernel driver doing its operations, which might be the problem you are experiencing (e.g. adjusting mic volume while recording).  Or another issue might be running this script too rapidly or having multiple instances of it running simultaneously.

Anyway, the proper solution might be to develop an actual kernel driver (I don't know how to do this, but I don't think it will be in python) which can co-exist smoothly with the existing kernel operations.  Feel free to experiment, sorry I cant be of much more help.
